### ISSUE TITLE
Detachable chat overlay as an option

### ISSUE THREAD

**Issue by sxiii** [Syncplay/syncplay#258]
It would be (maybe?) easier not to add text inside of the players (like in mpv) but rather to make a small always-on-top area for the text entry and optionally for the chat. And make somewhere checkbox that can show this overlay chat and maybe a hot corner to resize this field. That will add overlay chat support to all players at once, which is great. What do you think guys?

**Comment by Kliktopia** [comment]
By overlay are you talking about one with a solid background or a transparent or translucent background?

It is already the case that you can minimise the list of who is playing what and the playlist by dragging it smaller and that will leave just the notification window and chat box. I'm currently too busy to really do much Syncplay development, but if someone wanted to add an 'Always on top' option to the Window menu in Syncplay then they could do so and make a pull request (PR) - this can be accomplished using a WindowStaysOnTopHint window flag apparently.

Generally speaking, anyone who really cares about text-based chat should probably use mpv. Those who want a fancy overlay can try using Discord to do it. If you wanted anything more advanced then it is beyond my ability to implement, especially as I wouldn't be using (and therefore regularly testing) the functionality.

**Comment by Et0h** [comment]
Thanks for your suggestion. By overlay are you talking about one with a solid background or a transparent or translucent background?

It is already the case that you can minimise the list of who is playing what and the playlist by dragging it smaller and that will leave just the notification window and chat box. I'm currently too busy to really do much Syncplay development, but if someone wanted to add an 'Always on top' option to the Window menu in Syncplay then they could do so and make a pull request (PR) - this can be accomplished using a WindowStaysOnTopHint window flag apparently.

Generally speaking, anyone who really cares about text-based chat should probably use mpv. Those who want a fancy overlay can try using Discord to do it. If you wanted anything more advanced within Syncplay then it is beyond my ability to implement, especially as I wouldn't be using (and therefore regularly testing) the functionality.

**Comment by sxiii** [comment]
Hi @Et0h thanks for your comment. I think both transparent and solid-colored background can be used. I wasn't thinking about something fancy, just the ability to use some other players then MPV. Actually I'm getting used to MPV more, but I still think this addition could be quite interesting and helpful. 

Always-on-top can be also mostly as useful as a detachable chat windows. But the movies are mostly having dark or black background so we either need to make a "movie mode" with white text on the black background or, another way, make it possible to arrange colors for "background" and "foreground". As SyncPlay basically uses IRC, there are this basic color functionality in most IRC clients. It would be so nice to see something similar implemented in SyncPlay for the case of using overlay window chat.

Thanks!

**Comment by Et0h** [comment]
At the moment I'm generally wanting to avoid adding to the complexity of Syncplay too much, and the more features we have the more things people will have to learn how to use and the more code we will have to maintain for the foreseeable future on multiple platforms. As such, I have now added this suggestion to http://syncplay.pl/ideas/ under the heading: "Probably adds too much complexity to be worth it".

I have also added adding an 'Always on top' option to the Window menu in Syncplay to https://syncplay.pl/about/ideas/ under the heading of "Something we hope to get around to when we have time (volunteers welcome)". If someone wants to do this then they can make a pull request (PR) - this can be accomplished using a WindowStaysOnTopHint window flag apparently.
